lib/events.js

```diff
@@ -509,6 +509,17 @@ function once(emitter, name) {
       emitter.once('error', errorListener);
     }
 
-    emitter.once(name, eventListener);
+    if (typeof emitter.once === 'function' &&
```

Comment:
I could be going totally crazy but I feel like this function is incorrect and the test only passes because it runs through the EventEmitter path? I think you need something more like this?

```js
function once(emitter, name) {
  return new Promise((resolve, reject) => {
    if (typeof emitter.once === 'function' &&
        typeof emitter.removeListener === 'function') {
      const eventListener = (...args) => {
        if (errorListener !== undefined) {
          emitter.removeListener('error', errorListener);
        }
        resolve(args);
      };
      let errorListener;

      // Adding an error listener is not optional because
      // if an error is thrown on an event emitter we cannot
      // guarantee that the actual event we are waiting will
      // be fired. The result could be a silent way to create
      // memory or file descriptor leaks, which is something
      // we should avoid.
      if (name !== 'error') {
        errorListener = (err) => {
          emitter.removeListener(name, eventListener);
          reject(err);
        };

        emitter.once('error', errorListener);
      }

      emitter.once(name, eventListener);
      return;
    }

    if (typeof emitter.addEventListener === 'function') {
      // EventTarget does not have `error` event semantics like Node
      // EventEmitters, we do not listen to `error` events here.
      emitter.addEventListener(name, resolve, { once: true });
      return;
    }

    throw new ERR_INVALID_ARG_TYPE('emitter', 'EventEmitter', emitter);
  });
}
```

Reply:
I am not sure what the difference here - but this can be a brainfart on my behalf

Reply:
Your original PR still binds the `error` listener (and unnecessarily creates the `eventListener` function). It also creates a nested Promise that it then resolves so the original never resolves. I could also be missing something... but when I ran the code, it didn't seem to do what you're expecting.

Reply:
@apapirovski ah wow you're right!